// flow-start/node.js
class FlowStartNode {
  constructor(id, position) {
    this.id = id;
    this.type = 'flow-start';
    this.position = position;
    this.data = {
      title: '开始',
      description: '流程开始节点'
    };
    this.outputs = [{
      id: 'default',
      name: '下一步'
    }];
    this.inputs = [];
  }

  render(container) {
    container.classList.add('flow-node', 'start-node');
    
    const titleEl = document.createElement('div');
    titleEl.className = 'node-title';
    titleEl.textContent = this.data.title;
    container.appendChild(titleEl);
    
    const descEl = document.createElement('div');
    descEl.className = 'node-content';
    descEl.textContent = this.data.description;
    container.appendChild(descEl);
    
    return container;
  }

  getProperties() {
    return [
      {
        id: 'title',
        name: '标题',
        type: 'text',
        value: this.data.title
      },
      {
        id: 'description',
        name: '描述',
        type: 'textarea',
        value: this.data.description
      }
    ];
  }

  updateProperty(propertyId, value) {
    if (this.data.hasOwnProperty(propertyId)) {
      this.data[propertyId] = value;
      return true;
    }
    return false;
  }
}

// 暴露节点类
if (typeof module !== 'undefined') {
  module.exports = FlowStartNode;
}
